storage/remote/codec.go

```diff
@@ -372,9 +372,16 @@ func labelsToLabelsProto(labels labels.Labels) []prompb.Label {
 	result := make([]prompb.Label, 0, len(labels))
```

Comment:
You can be smarter here I think, I presume this is where the allocations are.

Reply:
:+1: Many were here, and some still in buildWriteRequest from making `[]prompb.Sample`. Latest update helps both of these.